subprojects/persistent-cache/src/main/java/org/gradle/cache/CleanupActionFactory.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.cache;
+
+public interface CleanupActionFactory {
```

Comment:
Why did this need to move to a non-`internal` package?  It looks like the implementers are all internal types.

Reply:
Not sure - might have been something I moved for some reason and then later moved back.  I should have caught this before asking for review.  I've moved it back.